openwisp_monitoring/check/base/models.py

```diff
@@ -75,27 +76,45 @@ def perform_check(self, store=True):
         return self.check_instance.check(store=True)
 
 
-if app_settings.AUTO_PING:
```

Comment:
I preferred the previous approach to be honest because previously we only registered the dispatcher when necessary (AUTO_PING=True). Now it is always registered (no matters the value of AUTO_PING) and hence executed on each post_save signal, causing the function to do nothing when AUTO_PING=False, note the difference.

Reply:
Makes sense but then I shall have to add a check in corresponding tasks. The reason being, the dispatcher shall get registered when the models are loaded. So, there is no option of mocking `AUTO_PING`, `AUTO_CONFIG_STATUS` it seems. Except for the tests, there is no real need of the check in the tasks.